crates/ruff_linter/src/rules/pyupgrade/rules/printf_string_formatting.rs

```diff
@@ -382,12 +380,11 @@ pub(crate) fn printf_string_formatting(checker: &mut Checker, expr: &Expr, right
     let mut num_positional_arguments = 0;
     let mut num_keyword_arguments = 0;
     let mut format_strings = Vec::with_capacity(strings.len());
-    for range in &strings {
+    for (range, flags) in &strings {
         let string = checker.locator().slice(*range);
-        let (Some(leader), Some(trailer)) = (leading_quote(string), trailing_quote(string)) else {
-            return;
-        };
-        let string = &string[leader.len()..string.len() - trailer.len()];
+        let quote_len = usize::from(flags.quote_len());
+        let string =
+            &string[(usize::from(flags.prefix_len()) + quote_len)..(string.len() - quote_len)];
```

Comment:
I agree that it's a common pattern, but the reason I held back from doing this was because it felt potentially quite confusing to me. `text_len()` would return `x + y` where `x` is the length of the prefixes and `y` is the length of the opening quotes -- right? But I feel like that's potentially a footgun, since you always want to use that for the opening of the string, and never for the closing of the string. If the API forces you to explicitly calculate it, then there's no footgun. Also:
- I wasn't sure what to call it -- `text_len()` to me implies that it would calculate the length of the contents of the string
- I'm not sure it would actually simplify the code much here, since we still need to do the conversion to `usize`

Writing this out, though, makes me think that maybe it would be an okay API if we gave it a clear enough name -- maybe something like `string_opener_len()`. WDYT?